### PR TITLE
Validate tag mappings

### PR DESCRIPTION
**Pull request by APaulic**
A script to map tags between 19139 and 19115-3.2018 schemas.

Outputs two overall files:
- Overall matches: Used as to map the data
- Overall unmatched: Used to identify data that is lost in the translate

The script (find_test_metadata_records.py):
1. Creates a list of every distinct tag used within a dump of GeoNetwork.
1. Finds the optimal set of metadata files to show an example of every tag. This is intended to produce a set of the minimal number of files to cover all tags used.
1. Extracts each tag and value from all of the example files found in the last step. This is done for both the 19139 and 19115-3.2018 records.
1. Compares the two files, 19139 & 19115-3.2018, for each metadata record. Any value that can not be matched from 19139 or 19115-3.2018 is added to unmatched_results.csv within the directory for the record. All matched elements are added with their mapped tags in matched_results.csv

The script (get_all_tag_values.py):
Searches through dump of GN and lists each tag and every distinct value for the tag.

### CODE REVIEW

**Comment by jonescc** [comment]
This the the mappings file I ended up with.

[confirmed_mappings.txt](https://github.com/aodn/utilities/files/4372316/confirmed_mappings.txt)

File: geonetwork/gn3-migration/tag-mapping/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+lxml==4.5.0
+pkg-resources==0.0.0
```
[the remark this answers is not in the document]
Comment (APaulic) [reply geonetwork/gn3-migration/tag-mapping/requirements.txt]: Sorry, its a [pip bug](https://github.com/pypa/pip/issues/4022) I used `pip freeze` to create the requirements file. I'll clean it up.

File: geonetwork/gn3-migration/tag-mapping/find_test_metadata_records.py [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+
+        if not matched:
+            tag_19139, value_19139 = [el.strip() for el in line.strip().split("||")]
+            unmatched_content_19139.append([tag_19139, value_19139])
+            if tag_19139 in TAGS_TO_SKIP:
+                mapped_tags[tag_19139] = None
+                with open(os.path.join(dest_dir, "confirmed_mappings.csv"), "a+") as mapping_file:
+                    mapping_file.write("%s,%s\n" % (tag_19139, SKIPPED))
+
+    # Compile list of unmatched 19115 values
```
[the remark this answers is not in the document]
Comment (APaulic) [reply geonetwork/gn3-migration/tag-mapping/find_test_metadata_records.py]: This script does operate under the assumption that data moves to one location in the new file and splitting/copying is manually mapped after execution. Skipping came in after this section was written and was not refactored. Unfortunately, this code wasn't a cure-all but a tool for one step in the process.

[the remark this answers is not in the document]
Comment (jonescc) [reply geonetwork/gn3-migration/tag-mapping/find_test_metadata_records.py]: OK.  Let's leave it as it is for the moment.  We've moved on from this now.  Would be good to fix the requirements issue though so if someone picks it up they can run it as described.

[the remark this answers is not in the document]
Comment (APaulic) [reply geonetwork/gn3-migration/tag-mapping/find_test_metadata_records.py]: The requirements.txt should now be fixed :+1:

**Pull request merged**